daemon/daemon.go

```diff
@@ -353,17 +353,43 @@ func (daemon *Daemon) restore() error {
 				// The container register failed should not be started.
 				return
 			}
+		}(c.container, c.registered)
```

Comment:
Does this "upper" go routine need to exist still now that actual container start operations are in the "lower" go routine (added below) with the `range restartContainers {` loop?

I tried to dig through `daemon.Register` and `daemon.generateNewName`--which seem to be the only external calls in the go routine now--to see if there is any reason for this to be **necessarily** handled per-container in a concurrent manner and nothing jumps out at me?